presto-main/src/main/java/io/prestosql/server/ui/WebUiAuthenticationModule.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server.ui;
+
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.prestosql.server.security.SecurityConfig;
+
+import java.util.List;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.airlift.configuration.ConfigBinder.configBinder;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public class WebUiAuthenticationModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        configBinder(binder).bindConfig(WebUiAuthenticationConfig.class);
+
+        installWebUiAuthenticator("form", new FormUiAuthenticatorModule());
+    }
+
+    private void installWebUiAuthenticator(String type, Module module)
+    {
+        install(webUiAuthenticator(type, module));
+    }
+
+    public static Module webUiAuthenticator(String type, Module module)
+    {
+        return new ConditionalWebUiAuthenticationModule(type, module);
+    }
+
+    private static class ConditionalWebUiAuthenticationModule
+            extends AbstractConfigurationAwareModule
+    {
+        private final String type;
+        private final Module module;
+
+        public ConditionalWebUiAuthenticationModule(String type, Module module)
+        {
+            this.type = requireNonNull(type, "type is null");
+            this.module = requireNonNull(module, "module is null");
+        }
+
+        @Override
+        protected void setup(Binder binder)
+        {
+            if (type.equals(getAuthenticationType())) {
+                install(module);
+            }
+        }
+
+        private String getAuthenticationType()
+        {
+            String authentication = buildConfigObject(WebUiAuthenticationConfig.class).getAuthentication();
+            if (authentication != null) {
+                return authentication;
+            }
+
+            List<String> authenticationTypes = buildConfigObject(SecurityConfig.class).getAuthenticationTypes().stream()
+                    .map(type -> type.toLowerCase(ENGLISH))
+                    .collect(toImmutableList());
+            if (authenticationTypes.contains("password")) {
+                return "form";
+            }
+            return authenticationTypes.stream()
```

Comment:
Why take the first one? What if none of them match a web UI authenticator type?

Reply:
The web ui only supports a single authenticator, so we choose the first on in the list, or if there are no authenticators, we use `form` since it handles that case.